test/e2e/framework/profile_gatherer.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+)
+
+func GatherApiserverCPUProfile(testBaseName string) error {
+	sshResult, err := SSH("wget -O - localhost:8080/debug/pprof/profile", GetMasterHost()+":22", TestContext.Provider)
```

Comment:
Does wgeting profile gives meaningful data? Normally profiler reads form it for 30s to tell anything.

Reply:
Yes it gives meaningful data (I checked it by plotting with `go tool pprof` against the binary file fetched by wget'ing). And the wget indeed takes ~30s to read the data.